Makefile

```diff
@@ -63,15 +71,25 @@ lib/libbase64.o: $(OBJS)
 	$(OBJCOPY) --keep-global-symbols=lib/exports.txt $@
 
 lib/config.h:
-	@echo "#define HAVE_AVX2   $(HAVE_AVX2)"    > $@
-	@echo "#define HAVE_NEON32 $(HAVE_NEON32)" >> $@
-	@echo "#define HAVE_NEON64 $(HAVE_NEON64)" >> $@
-	@echo "#define HAVE_SSSE3  $(HAVE_SSSE3)"  >> $@
-	@echo "#define HAVE_SSE41  $(HAVE_SSE41)"  >> $@
-	@echo "#define HAVE_SSE42  $(HAVE_SSE42)"  >> $@
-	@echo "#define HAVE_AVX    $(HAVE_AVX)"    >> $@
+	@echo "#define HAVE_AVX2                  $(HAVE_AVX2)"                   > $@
+	@echo "#define HAVE_NEON32                $(HAVE_NEON32)"                >> $@
+	@echo "#define HAVE_NEON64                $(HAVE_NEON64)"                >> $@
+	@echo "#define HAVE_SSSE3                 $(HAVE_SSSE3)"                 >> $@
+	@echo "#define HAVE_SSE41                 $(HAVE_SSE41)"                 >> $@
+	@echo "#define HAVE_SSE42                 $(HAVE_SSE42)"                 >> $@
+	@echo "#define HAVE_AVX                   $(HAVE_AVX)"                   >> $@
+	@echo "#define HAVE_FAST_UNALIGNED_ACCESS $(HAVE_FAST_UNALIGNED_ACCESS)" >> $@
+
+lib/table_generator: lib/table_generator.o
+	$(CC) $(CFLAGS) -o $@ $^
+
+lib/tables.h: lib/table_generator.c
+	@$(MAKE) -f $(THIS_FILE) lib/table_generator
+	./lib/table_generator > $@
+
```

Comment:
Having `lib/tables.h` in source control is fine, the file doesn't need to be regenerated every time the library is rebuilt.

I think the machinery to force a dependency on `lib/table_generator.c` is a bit too complex. Since `lib/table_generator` already depends on `lib/table_generator.o`, won't the binary always get rebuilt when `lib/table_generator.c` is changed?

If someone wants to regenerate `lib/tables.h`, they can delete that file and it will be rebuilt from source using whatever fresh version of `lib/table_generator.c` is available. I think that's sufficiently straightforward and clean.